### PR TITLE
Update commonly used x86 regs list and add mtrr support

### PR DESCRIPTION
**Pull request by aisaila**

### CODE REVIEW

**Comment by aisaila** [comment]
@tklengyel can you please take a look at this pull request when you have the time?

Thanks

File: libvmi/libvmi.h [hunk 1]
```diff
@@ -481,6 +481,7 @@ typedef uint64_t reg_t;
 #define SPSR_EL1    SPSR_SVC
 #define TCR_EL1     TTBCR
 
+typedef struct hvm_hw_mtrr mtrr_regs_t;
```
Comment (tklengyel) [inline libvmi/libvmi.h]: This is no good. Are you pulling this in from a Xen header? What about non-Xen systems?

Comment (aisaila) [reply libvmi/libvmi.h]: I've amended the pull request with a custom libvmi mtrr regs structure and moved the xen specific structure use in the xen driver.

**Comment by smaresca** [comment]
@aisaila please resolve conflicts and edit the title to be more descriptive

**Comment by aisaila** [comment]
I have updated the title and resolved the conflicts. The PR can be merged.

**Comment by tklengyel** [comment]
@libvmi-jenkins This is OK to test

**Comment by aisaila** [comment]
Can this be merged or is there something else that needs to be changed?

**Comment by tklengyel** [comment]
LGTM

**Comment by tklengyel** [comment]
@smaresca if you have no further issue with this PR hit merge when you get a chance

**Pull request merged**